sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -48,14 +117,32 @@ namespace Azure { namespace Core { namespace Http {
       // insert headers
       for (auto header : headers)
       {
-        // TODO: check result is not null or trow
         headerList = curl_slist_append(headerList, (header.first + ":" + header.second).c_str());
+        if (headerList == NULL)
+        {
+          throw;
+        }
+      }
+
+      if (isUploadRequest())
+      {
+        // set expect header for PUT and POST request. This disables libcurl to send only headers
+        // and expect sever to return a Continue respond before posting body
+        headerList = curl_slist_append(headerList, "Expect:");
+        // inf header for payload size
+        auto requestStream = this->m_request.GetBodyStream();
+        auto size = requestStream != nullptr ? requestStream->Length()
```

Comment:
can you use const more often, `auto const` in this case? It is easier to read the code knowing that the variable is not going to change value up until the end of the function.

Reply:
suggestion